public/js/reducers/sources.js

```diff
@@ -28,9 +29,14 @@ function update(state = initialState, action) {
       );
 
     case constants.SELECT_SOURCE:
-      return state.merge({
-        selectedSource: action.source
-      });
+      return state
+        .merge({ selectedSource: action.source })
+        .set("tabs", addSourceToTabList(state, fromJS(action.source)));
```

Comment:
why not `state.merge({ selectedSource: action.source, tabs: addSourceToTabList(...) })` ? You can do it in the same `merge` rather than another `set`.